scarb/src/internal/serdex.rs

```diff
@@ -1,9 +1,73 @@
-use anyhow::Result;
+use anyhow::{anyhow, Result};
 use camino::{Utf8Path, Utf8PathBuf};
 use serde::{Deserialize, Serialize};
 
 use crate::internal::fsx;
 
+const MERGE_STRATEGY_KEY: &str = "merge-strategy";
+
+#[derive(Debug, Default)]
+pub enum MergeStrategy {
+    #[default]
+    Override,
+    Merge,
+}
+
+impl TryFrom<&str> for MergeStrategy {
+    type Error = anyhow::Error;
+    fn try_from(value: &str) -> std::result::Result<Self, Self::Error> {
+        match value {
+            "override" => Ok(Self::Override),
+            "merge" => Ok(Self::Merge),
+            _ => Err(anyhow!(
+                "invalid merge strategy: {}, must be one of `merge`, `override`",
+                value
+            )),
+        }
+    }
+}
+
+pub fn toml_merge_apply_strategy<'de, T, S>(target: &T, source: &S) -> Result<T>
```

Comment:
doc, you can use excerpts from the design doc

Reply:
Sure, I will add it as soon as it will be accepted :)

Reply:
I meant, document this function :)

Reply:
Yep, I meant it too :D